src/GitHub.App/Services/GitClient.cs

```diff
@@ -394,7 +394,7 @@ public Task<bool> IsModified(IRepository repository, string path, byte[] content
                 baseCommit = repo.Lookup<Commit>(baseSha);
                 if (baseCommit == null)
                 {
-                    return null;
+                    throw new NotFoundException($"Couldn't find {baseSha} after fetching from {baseCloneUrl}:{baseRef}.");
```

Comment:
So I understand why you'd want the details of the shas in the log, but this error message is incredibly cryptic, there is no way anyone is going to know what is going on with an error like this, and this is going to be displayed directly to a user where it's going to be completely useless.

We should have a message for the user, and another with more details for the log. If the log is only showing the message and stacktrace, we can set InnerException with the log details and have the outer exception be for the user.

Reply:
Also, are all the callers of this method expecting an exception to be thrown and handling it properly?

Reply:
Callers of this method previously turned the null into an exception themselves. I've also tested it by making this method always throw to check that nothing bad happens. The exception message is displayed as expected.

Reply:
I wonder what the user facing message should say? I suspect that for some reason, fetch doesn't always work or there is some race condition. I guess we could suggest thry try again or maybe refresh. I'll investagate...

Reply:
Yeah, an indication of what the user action should be in the case of this error would be great!

Reply:
Something like this.

![image](https://user-images.githubusercontent.com/11719160/29712192-7e0a37ce-8990-11e7-8392-c2134c664b01.png)

Or maybe simply, "Please check your network connection and try again."?

Reply:
"Couldn't find merge base. Please check your network connection and try again."

I think this is maybe the right balance of revealing what actually went wrong, but giving the user a hint that has a good chance of fixing it.

Reply:
Here we go:
![image](https://user-images.githubusercontent.com/11719160/29712641-c7adaf12-8992-11e7-8123-99c75b643acd.png)

Reply:
So "merge base" may be meaningless to a user, and try again might not be helpful either if the action that the user performed to get here isn't available from this screen.

What is the action that failed here? Opening a file? Showing the PR details? The first part of the message should say what was the action that failed.

What is try again? Refresh? Click on the PR button in the toolbar and then clicking on the PR again? The second part should say what the user can do on this screen to attempt to make it work.

Assuming it was the PR that failed to load, then we should probably say something like

`The Pull Request failed to load. Please check your network connection and click refresh to try again.`

Adjust accordingly for the failed action.

Reply:
Okay, I've chained it to variants of your suggestion above.

My concern was that we hide the real issue behind a workaround. I'm still keen to get to the bottom of what's going wrong. My hunch is that it's a race condition in gitlib2sharp. I'll see if I can reproduce the issue with some standalone code... 😄

Reply:
Yes, we don't want to hide the issue in a workaround. If you're concerned about people not reporting it, you can add "If this issue persists, please let us know at support@github.com"

Reply:
Done. Also confirmed that nested exceptions appear in logs.